src/framework/filterControl/forms/MultipleSelectFilterForm.jsx

```diff
@@ -19,16 +19,17 @@ export default class MultipleSelectFilterForm extends Component {
 
     this.onCheckBoxClick = this.onCheckBoxClick.bind(this);
     this.onClickAddButton = this.onClickAddButton.bind(this);
-    this.options = this.props.filterOption.comparisonParameters.oneOfOptions;
+    this.options = props.filterOption.comparisonParameters.oneOfOptions;
 
     let selectedOptions;
-
     if (this.props.comparisonValue === undefined) {
       selectedOptions = (new Array(this.options.length)).fill(false);
     } else {
-      selectedOptions = this.options.map(option => (
-        this.props.comparisonValue.indexOf(option) !== -1
-      ));
+      selectedOptions = this.options.map(option =>
+        props.comparisonValue.reduce(
+          (previous, element) =>
+            element.value === option.value || previous, false)
+      );
```

Comment:
The line wrap is a little off here, remember opening and closing bracket on the same level, then if multilines try to wrap each parameter. So something like 
```
props.comparisonValue.reduce(
  (foundValue, element) => element.value === option.value || foundValue,
  false
)
```

Plus the reduce implementation is neat but a find would also do, as you look for a specific object. So:
```
selectedOptions = this.options.map(option =>
  Boolean(props.comparisonValue.find(
    element => element.value === option.value)
  )
);
```